.github/actions/generate-attestations/src/main.ts

```diff
@@ -0,0 +1,51 @@
+import * as core from "@actions/core";
+import fs from "fs";
+import path from "path";
+import { writeAttestations } from "./attestation";
+import { resolvePathInput } from "./utils";
+
+export async function run(): Promise<void> {
+  try {
+    const wd = process.env[`GITHUB_WORKSPACE`] || "";
+
+    // SLSA subjects layout file.
+    const slsaOutputs = core.getInput("slsa-outputs-file");
+    const safeSlsaOutputs = resolvePathInput(slsaOutputs, wd);
+    core.debug(`Using SLSA output file at ${safeSlsaOutputs}!`);
+
+    // Predicate.
+    const predicateFile = core.getInput("predicate-file");
+    const safePredicateFile = resolvePathInput(predicateFile, wd);
+    core.debug(`Inputs: Predicate file ${safePredicateFile}!`);
+
+    // Predicate type
+    const predicateType = core.getInput("predicate-type");
+    core.debug(`Inputs: Predicate type ${predicateType}!`);
+
+    // Attach subjects and generate attestation files
+    const outputFolder = core.getInput("output-folder");
+    const attestations = await writeAttestations(
+      safeSlsaOutputs,
+      predicateType,
+      safePredicateFile
+    );
+
+    // Write attestations
+    fs.mkdirSync(outputFolder, { recursive: true });
+    for (const att in attestations) {
+      const outputFile = path.join(outputFolder, att);
+      const safeOutput = resolvePathInput(outputFile, wd);
+      fs.writeFileSync(safeOutput, attestations[att]);
+    }
+
+    core.setOutput("output-folder", outputFolder);
```

Comment:
curious why this is needed if it's the same as the `output-folder` input

Reply:
output-folder is an optional input with a default, so that callers would know the resolved one as output. It's not necessary though, since the caller would know what it used as input/default.

Reply:
resolved below, making it a required one, since it will likely need to be randomized to avoid multiple calls to an SRW